contents/blog/missing-recordings.md

```diff
@@ -0,0 +1,73 @@
+
+We decided to use the second metric because it would let us strive to 0% missing recordings. If you're curious, you can read the conversation that led to this decision in [this GitHub issue](https://github.com/PostHog/posthog/issues/5478).
+
+In the first run of the metric, it showed that we we're missing ~15% of recordings across [PostHog Cloud](/pricing) users.
+
+## Checking Sentry for clues
+
+Now that we had a metric, we started hunting for clues. What was causing the missing recordings?
+
+The first place we looked was Sentry. There were a couple of errors in the ingestion pipeline that seemed suspicious. The most notable was a "Max data upload size exceeded" error that was firing a few thousand times a day. This seemed like it could be an obvious cause of missing recordings. If some data was not being uploaded because it was too big, it would make sense that we could not get a 'full snapshot'.
+
+To figure out what was causing this error, we looked at data that had been uploaded that was just below our size threshold of 20mb. It looked like large data URLs of images were the cause.
+
+To test this out, we made a filter that would remove data URLs from any payload that exceeded a 5mb threshold, and it would replace the image with an obvious placeholder.
+
+![Recording Page with SVGs](https://user-images.githubusercontent.com/4813045/137223852-eeb4273e-d81c-49ca-a115-02b9cd0979e1.png)
+
+One issue we faced was that we didn't want this feature to significantly increase the size of `posthog-js`, but we would have to include the placeholder image in the bundle. After some discussion, we landed on the idea of using a very small SVG that could easily scale to the size of the image being replaced. You can read more about the discussion that landed us here in [the PR itself](https://github.com/PostHog/posthog-js/pull/317).
+
+
+After deploying the fix, the Sentry error went away, but the missing recording only went down a few percent.
+
+## Looking deeper
+
+What was causing the rest of the missing recordings? With no more clues, it came time to dig through the mechanics of the code.
+
+When a recording started, we took a 'full snapshot' of the page, and from that point forward, we would send up only the data that changed in the recording. For some reason, the missing recordings would often have the later data, but no full snapshot.
+
+Why were the full snapshots missing? Presumably, if the later data was being sent up, the snapshot code must've already run. To further solidify this line of thinking, the code to determine the changes to the page depends entirely on the snapshot existing in memory, so the system must've already taken a full snapshot.
+
+After a day or so of pounding our heads on the keyboard, we realized that recordings were being split when a user was inactive for 30 minutes. But when the split happened, we were not taking a new full snapshot.
```

Comment:
I feel like there's some missing detail here that would really round off the narrative. What steps / dead ends did we / you go down while you were pounding heads on keyboards?